commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/v1_11_1/CPDisplayLayoutUpgradeProcess.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.product.internal.upgrade.v1_11_1;
+
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.Layout;
+import com.liferay.portal.kernel.service.LayoutLocalService;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+/**
+ * @author Riccardo Ferrari
+ */
+public class CPDisplayLayoutUpgradeProcess extends UpgradeProcess {
+
+	public CPDisplayLayoutUpgradeProcess(
+		LayoutLocalService layoutLocalService) {
+
+		_layoutLocalService = layoutLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		Statement s = connection.createStatement(
+			ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
+
+		ResultSet r = s.executeQuery(_SELECT_CPDISPLAYLAYOUT_SQL);
+
+		while (r.next()) {
+			long cpDisplayLayoutId = r.getLong("CPDisplayLayoutId");
+
+			long groupId = r.getLong("groupId");
+
+			String layoutUuid = r.getString("layoutUuid");
+
+			Layout layout = _getLayout(groupId, layoutUuid);
+
+			if (layout == null) {
+				if (_log.isDebugEnabled()) {
+					_log.debug(
+						String.format(
+							"Removing orphan CPDisplayLayout entry id: %s",
+							cpDisplayLayoutId));
+				}
+
+				runSQL(
+					String.format(
+						_DELETE_CPDISPLAYLAYOUT_SQL, cpDisplayLayoutId));
```

Comment:
This..... is wrong, isn't it?
`"DELETE FROM CPDisplayLayout where '%s'".format(id)`
yields something like
`"DELETE FROM CPDisplayLayout where '1234'"`.

We need to name the column we're `where`ing by, and I don't think we should be quoting the ID, either.

Reply:
ouch this was bad, I guess I should get some rest